libsolidity/analysis/TypeChecker.cpp

```diff
@@ -709,6 +701,23 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 					m_errorReporter.typeError(6617_error, _identifier.location, "The suffixes _offset and _slot can only be used on non-constant storage variables.");
 					return size_t(-1);
 				}
+				else if (var && var->value() && !var->value()->annotation().type && dynamic_cast<Literal const*>(var->value().get()) == nullptr)
```

Comment:
What do you think about:
```suggestion
				else if (var && var->value() && !var->value()->annotation().type && !dynamic_cast<Literal const*>(var->value().get()))
```

Reply:
The file already has `== nullptr` pattern. So maybe this change is not needed.

Reply:
Changed